plugin-volume/volumepopup.cpp

```diff
@@ -197,8 +208,23 @@ void VolumePopup::openAt(QPoint pos, Qt::Corner anchor)
 
 void VolumePopup::handleWheelEvent(QWheelEvent *event)
 {
-    m_volumeSlider->setSliderPosition(m_volumeSlider->sliderPosition()
-            + (event->angleDelta().y() / QWheelEvent::DefaultDeltasPerStep * m_volumeSlider->singleStep()));
+    static int _delta = 0; // for high-resolution mice and touchpad scrolling
+
+    int delta = event->angleDelta().y();
+    if ((_delta ^ delta) < 0)
+        _delta = delta; // the wheel direction is reversed
+    else
+        _delta += delta;
+    if (qAbs(_delta) >= QWheelEvent::DefaultDeltasPerStep) {
+        m_volumeSlider->setSliderPosition(m_volumeSlider->sliderPosition()
+                + (_delta / QWheelEvent::DefaultDeltasPerStep * m_volumeSlider->singleStep()));
+        _delta = 0;
```

Comment:
For the same reason a above, I suggest to write this as `_delta = _delta % QWheelEvent::DefaultDeltasPerStep`

Reply:
It would be very wrong. → https://github.com/lxqt/lxqt-panel/pull/1856#issuecomment-1381367296 → 2 and especially 4